zipserver/zip_handler.go

```diff
@@ -91,28 +88,26 @@ func zipHandler(w http.ResponseWriter, r *http.Request) error {
 		return err
 	}
 
-	if keyBusy(key) {
+	hasLock := tryLockKey(key)
```

Comment:
we should defer `releaseKey` immediately after this line, any errors that may happen below between the release calls would cause the lock to get stuck